### DISCUSSION TITLE
Question: "type first" declaration reasons?

### DISCUSSION THREAD

**Discussion by renlite** [c3lang/c3c#1485] [General]
Hi, 

first, I like the syntax of C3.

Why did you choose to use the C-stlye declaration for C3?

```C
fn void Point.add(Point* p, int x)
{
    p.x += x;
}

fn bool testme(int a, double... x, double rate = 1.0) 
{ 
   /* ... */ }
}
```

IMO the Pascal-style syntax has better readability. 

```C
fn Point.add(p *Point, x int) -> void
{
    p.x += x;
}

fn testme(a int, x ...double, rate double = 1.0)  -> bool
{ 
   /* ... */ }
}
```

**Comment by Caleb-o** [comment]
Hello o7

A lot of the design of the language usually comes down to similarity and familiarity with C. So while some syntax has been changed, it is still pretty close C's syntax. C3 is intended to be a better C, essentially adding new features and semantics on top of C while keeping as close to it as possible, to make migration and working along side C easier.

**Comment by lerno** [comment]
Like Caleb is saying, the intent is to keep close to C syntax as much as possible. The `fn` is there because there is a lot of added value to having it. There is advantages of "type last" too, but mostly in the code now aligning better, which is a rather limited advantage.

Languages that have "type last" functions usually do the same for variable declarations and use this to simplify the grammar. C3 doesn't do that (variable declaration with type last), so just doing it for functions would also be fairly inconsistent.

This is not to say that "type last" is bad in any way. It's just too far from C syntax.

**Comment by renlite** [comment]
Yes, with "type last" I also meant the variable declaration to be consistent. I see the compatibility with C has a higher priority.

**Reply by lerno** [reply]
Yes. Being close to C for C programmers is a major priority for the language.
